qp_target_gene/trimming.py

```diff
@@ -0,0 +1,107 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import join
+from functools import partial
+from h5py import File
+
+from qiita_client import ArtifactInfo
+
+from qp_target_gene.split_libraries.util import generate_demux_file
+from qiita_files.demux import fetch
+from qiita_files.format.fasta import format_fasta_record
+from qiita_files.format.fastq import format_fastq_record
+
+
+def generate_trimming(filepaths, out_dir, parameters):
+    """Generates the pick_closed_reference_otus.py command
+
+    Parameters
+    ----------
+    filepaths : list of str
+        The demux filepaths
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    str
+        The pick_closed_reference_otus.py command
+        The output directory
+    """
+    length = parameters['length']
+
+    id_fmt = (b"%(sample)s_%(idx)d orig_bc=%(bc_ori)s new_bc=%(bc_cor)s "
+              b"bc_diffs=%(bc_diff)d")
+    pd = partial(join, out_dir)
+    fna_fp = pd('seqs.fna')
+    fastq_fp = pd('seqs.fastq')
+    fna_fh = open(fna_fp, 'w')
+    fastq_fh = open(fastq_fp, 'w')
+    for f in filepaths:
```

Comment:
Why do you have a list of filepaths? Artifacts will only contain a single HDF5 demux file (by definition)